File: backend/src/modules/insights/domain/attention-item.ts
export const ATTENTION_TYPES = [
  'LESSONS_WITHOUT_ATTENDANCE',
  'OVERDUE_UNGRADED_ACTIVITIES',
  'ACTIVITIES_AWAITING_GRADE',
  'CONTENTS_IN_PROGRESS',
  'STUDENTS_PENDING_SUBMISSION',
  'ABSENT_ON_ACTIVITY_LESSON',
  'EXCESS_ABSENCES',
  'ACTIVITIES_WITHOUT_SCORE',
] as const;

export type AttentionType = (typeof ATTENTION_TYPES)[number];

export const ATTENTION_SEVERITIES = ['high', 'medium', 'low'] as const;

export type AttentionSeverity = (typeof ATTENTION_SEVERITIES)[number];

export type AttentionItemFilters = {
  academicYearId?: string;
  classId?: string;
} & Record<string, unknown>;

/** Pendência acionável do Dashboard — ver docs/ARCHITECTURE.md §9 (Insights Engine). */
export type AttentionItem = {
  id: string;
  type: AttentionType;
  severity: AttentionSeverity;
  title: string;
  message: string;
  count: number;
  actionRoute: string;
  filters: AttentionItemFilters;
};

File: frontend/lib/domain/entities/attention_item.dart
class AttentionItem {
  const AttentionItem({
    required this.id,
    required this.type,
    required this.severity,
    required this.title,
    required this.message,
    required this.count,
    this.actionRoute,
    this.filters = const {},
  });

  final String id;
  final String type;
  final String severity;
  final String title;
  final String message;
  final int count;
  final String? actionRoute;
  final Map<String, dynamic> filters;
}

class DashboardSummary {
  const DashboardSummary({
    required this.totalAttentionItems,
    required this.totalPendingActions,
    required this.bySeverity,
  });

  final int totalAttentionItems;
  final int totalPendingActions;
  final Map<String, int> bySeverity;
}

class DashboardResponse {
  const DashboardResponse({required this.attentionItems, required this.summary});

  final List<AttentionItem> attentionItems;
  final DashboardSummary summary;
}
